### ISSUE TITLE
ImportError: No module named 'commands'

### ISSUE THREAD

**Issue by lwencel** [NervanaSystems/aeon#49]
Aeon does not build with python3. 

```
Processing /root/aeon_build
     Complete output from command python setup.py egg_info:
     Traceback (most recent call last):
       File "<string>", line 1, in <module>
       File "/tmp/pip-lffpu_zk-build/setup.py", line 6, in <module>
         import commands
     ImportError: No module named 'commands'
```

Log: `http://jenkins.sd.nervana.ai:8080/view/NGraph/job/NGraph_Docker_Images_Update/81/console`

**Comment by uberloss** [comment]
This is caused by cmake 2.8.* finding pythonLibs in version <=3.3 while 3.4 is installed. This can be fixed for now by applying a patch. It will be officially patched in never release. 

```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 93ae83f..90ce503 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -52,6 +52,7 @@ list(REMOVE_ITEM OpenCV_LIBRARIES opencv_ts)
 find_package(CURL REQUIRED)
 include_directories(${CURL_INCLUDE_DIRS})

+set(Python_ADDITIONAL_VERSIONS 3.4)
 find_package(PythonLibs)
 find_package(PythonInterp)

diff --git a/src/setup.py.in b/src/setup.py.in
index e10e152..febf7db 100644
--- a/src/setup.py.in
+++ b/src/setup.py.in
@@ -3,7 +3,8 @@ import sys
 from distutils.core import setup, Extension
 from distutils.sysconfig import get_python_inc
 import subprocess
-import commands
+if sys.version_info[0] < 3:
+    import commands
 import numpy

 # Parallel build from http://stackoverflow.com/questions/11013851/speeding-up-build-process-with-distutils
@@ -28,21 +29,34 @@ lib_dirs = ""
 libs = ""
 include_dirs = ""

-status, tmp = commands.getstatusoutput("pkg-config --cflags opencv")
+if sys.version_info[0] < 3:
+    status, tmp = commands.getstatusoutput("pkg-config --cflags opencv")
+else:
+    status, tmp = subprocess.getstatusoutput("pkg-config --cflags opencv")
 if status != 0:
     print("required package 'opencv' not found")
     exit()
 include_dirs += tmp
-lib_dirs += commands.getoutput("pkg-config --libs-only-L opencv")
-libs += commands.getoutput("pkg-config --libs-only-l opencv")
+if sys.version_info[0] < 3:
+    lib_dirs += commands.getoutput("pkg-config --libs-only-L opencv")
+    libs += commands.getoutput("pkg-config --libs-only-l opencv")

-status, tmp = commands.getstatusoutput("pkg-config --cflags sox")
+    status, tmp = commands.getstatusoutput("pkg-config --cflags sox")
+else:
+    lib_dirs += subprocess.getoutput("pkg-config --libs-only-L opencv")
+    libs += subprocess.getoutput("pkg-config --libs-only-l opencv")
+
+    status, tmp = subprocess.getstatusoutput("pkg-config --cflags sox")
 if status != 0:
     print("required package 'sox' not found")
     exit()
 include_dirs += tmp
-lib_dirs += commands.getoutput("pkg-config --libs-only-L sox")
-libs += commands.getoutput("pkg-config --libs-only-l sox")
+if sys.version_info[0] < 3:
+    lib_dirs += commands.getoutput("pkg-config --libs-only-L sox")
+    libs += commands.getoutput("pkg-config --libs-only-l sox")
+else:
+    lib_dirs += subprocess.getoutput("pkg-config --libs-only-L sox")
+    libs += subprocess.getoutput("pkg-config --libs-only-l sox")
 libs += ' -lcurl'

 os.environ["CC"] = "clang++"
```

https://docs.python.org/2/library/commands.html
https://bugs.launchpad.net/ubuntu/+source/cmake/+bug/1323559

**Comment by jennifermyers** [comment]
can we push the tag/fix that @DawnStone is using in private-aeon to public aeon so she can switch back to using public aeon?

**Comment by lwencel** [comment]
@rkimballn1 how to merge to public master?

**Comment by pzelazko-intel** [comment]
fixed in #50
